src/common/InputRowRadioButtons.js

```diff
@@ -0,0 +1,154 @@
+/* @flow strict-local */
+import React, { useCallback, useRef, useMemo, useEffect } from 'react';
+import type { Node } from 'react';
+import { View } from 'react-native';
+import invariant from 'invariant';
+import { CommonActions } from '@react-navigation/native';
+import { BRAND_COLOR, createStyleSheet } from '../styles';
+
+import Touchable from './Touchable';
+import ZulipText from './ZulipText';
+import { IconRight } from './Icons';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+
+type Item<TKey> = $ReadOnly<{|
+  key: TKey,
+  title: string,
+  subtitle?: string,
+|}>;
+
+type Props<TItemKey> = $ReadOnly<{|
+  /**
+   * Component must be under the stack nav with the selectable-options screen.
+   *
+   * Pass this down from props or `useNavigation`.
+   */
+  navigation: AppNavigationProp<>,
+
+  /** What the setting is about, e.g., "Theme". */
+  label: string,
+
+  /** What the setting is about, in a short sentence or two. */
+  description?: string,
+
+  /** The current value of the input. */
+  valueKey: TItemKey,
+
+  /**
+   * The available options to choose from, with unique `key`s, one of which
+   * must be `valueKey`. items: $ReadOnlyArray<Item<TItemKey>>,
+   */
+  items: $ReadOnlyArray<Item<TItemKey>>,
+
+  onValueChange: (newValue: TItemKey) => void,
+|}>;
+
+/**
+ * A form-input row for the user to make a choice, radio-button style.
+ *
+ * Shows the current value (the selected item), represented as the item's
+ * `.title`. When tapped, opens the selectable-options screen, where the
+ * user can change the selection or read more about each selection.
+ */
+// This has the navigate-to-nested-screen semantics of NestedNavRow,
+// represented by IconRight. NestedNavRow would probably be the wrong
+// abstraction, though, because it isn't an imput component; it doesn't have
+// a value to display.
+export default function InputRowRadioButtons<TItemKey: string>(props: Props<TItemKey>): Node {
+  const { navigation, label, description, valueKey, items, onValueChange } = props;
+
+  const screenKey = useRef(
+    `selectable-options-${Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(36)}`,
```

Comment:
nit: Probably this should be a helper function with a name, for the sake of both brevity and clarity. Which basically means take the existing `eg.randString` and put it somewhere in the non-test code.